### PR TITLE
BaSP-MR-55: Create Trainer List/Delete table

### PR DESCRIPTION
**Pull request by marcohzo**
Establishes a connection with the Trainer database, enabling the functionality to view the list of trainers, delete existing trainers, and lays the groundwork for future updates to allow editing and addition of new trainers.

Changes made:

Established a secure connection with the Trainer database.
Implemented the feature to display the list of trainers, providing easy access to their details.
Added the functionality to delete trainers from the database, facilitating data management.

### CODE REVIEW

File: src/Components/Trainers/index.js [hunk 1]
```diff
@@ -1,11 +1,57 @@
...
+        <button onClick={handleCloseModal}>Close</button>
+      </div>
+    );
+  };
+  const deleteItem = async (id) => {
+    try {
+      await fetch(`https://nai-megarocket-server.vercel.app/api/trainers/${id}`, {
+        method: 'DELETE'
+      });
+
```
Comment (AleMac17) [inline src/Components/Trainers/index.js]: We should not leave empty lines

**Review by AleMac17 — CHANGES_REQUESTED** [review]

File: src/Components/Trainers/index.js [hunk 2]
```diff
@@ -1,11 +1,57 @@
...
+/* import Modal from './Modal'; */
+import { useEffect, useState } from 'react';
+
+const Trainers = () => {
+  const [trainers, setTrainers] = useState([]);
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [deleteModalMessage, setDeleteModalMessage] = useState('');
+
+  const getTrainers = async () => {
+    const response = await fetch('https://nai-megarocket-server.vercel.app/api/trainers');
```
Comment (lorenzomauro59) [inline src/Components/Trainers/index.js]: it is recommended to use the variable from the server's .env file
```
      const response = await fetch(`${process.env.REACT_APP_API_URL + '/trainers'}`);

```

**Review by lorenzomauro59 — CHANGES_REQUESTED** [review]

File: src/Components/Trainers/index.js [hunk 3]
```diff
@@ -1,11 +1,57 @@
 import styles from './trainers.module.css';
+import Table from './Table';
+/* import Modal from './Modal'; */
```
Comment (EmaBaldor) [inline src/Components/Trainers/index.js]: Is recommended not to leave these comments

File: src/Components/Trainers/index.js [hunk 4]
```diff
@@ -1,11 +1,57 @@
...
+import { useEffect, useState } from 'react';
+
+const Trainers = () => {
+  const [trainers, setTrainers] = useState([]);
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [deleteModalMessage, setDeleteModalMessage] = useState('');
+
+  const getTrainers = async () => {
```
Comment (EmaBaldor) [inline src/Components/Trainers/index.js]: You can add try and catch here too

File: src/Components/Trainers/Table/index.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+  return (
+    <div>
+      <table className={styles.tableContainer}>
+        <thead>
+          <tr>
+            <th>First Name</th>
+            <th>Last Name</th>
+            <th></th>
+            <th>Actions</th>
```
Comment (EmaBaldor) [inline src/Components/Trainers/Table/index.js]: *optional: you can remove the blank columns and unify the actions column. 
Example:
ee

File: src/Components/Trainers/index.js (hunk 4 above)
Comment (marcohzo) [reply src/Components/Trainers/index.js]: Thank you for your review. The changes have been completed and it's ready!

File: src/Components/Trainers/index.js (hunk 2 above)
Comment (marcohzo) [reply src/Components/Trainers/index.js]: Thank you for your review. The changes have been completed and it's ready!

File: src/Components/Trainers/index.js (hunk 1 above)
Comment (marcohzo) [reply src/Components/Trainers/index.js]: Thank you for your review. The changes have been completed and it's ready!

**Comment by fernandofernandezuy** [comment]
[[Super-Admin] List/Delete table](https://trello.com/c/mw2Yf6WM/48-super-admin-list-delete-table)

**Review by lorenzomauro59 — APPROVED** [review]

File: src/Components/Trainers/index.jsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      setTrainers(data.data);
+    } catch (error) {
+      setDeleteModalMessage('Error occurred while retrieving the trainers');
+      setShowDeleteModal(true);
+    }
+  };
+  useEffect(() => {
+    getTrainers();
+  }, []);
+  const DeleteModal = () => {
```
Comment (LucianoAlarcon) [inline src/Components/Trainers/index.jsx]: Please, rename this function to `deleteModal`

**Review by AleMac17 — APPROVED** [review]

**Review by LucianoAlarcon — CHANGES_REQUESTED** [review]

**Review by LucianoAlarcon — APPROVED** [review]

**Comment by nazarenapantanetti** [comment]
Tested and approved!

**Pull request closed without merging**